### ISSUE TITLE
Refactor node manager

### ISSUE THREAD

**Issue by adambabik** [status-im/status-go#403]
# Problem

The logic behind starting and stopping a node is very cryptic and hard to follow. It also prevents us from properly finishing other tasks like async send calls. 

# Implementation

1. Stopping a node should first cancel all background tasks and wait for them to finish (e.g. jail's `sendAsync` handlers).
2. Calling for `isNodeAvailable()` in `NodeManager` methods is removed as a caller has no control now. If the caller needs a running node, it should make sure it is before calling a given method.
3. TBD

# Acceptance Criteria

1. `Node` and `NodeManager` are greatly simplified and easy to reason about,
2. Unit tests coverage is at least 80%,
3. There are no race conditions when running unit tests with `-race`.
4. `Node` and `NodeManager` are thread-safe.

**Comment by JekaMas** [comment]
I think this is related to [258](github.com/status-im/status-go/issues/258)

**Comment by adambabik** [comment]
@JekaMas you're correct. I think #258 should be incorporated into this issue.

**Comment by themue** [comment]
As seen in PR #420 the `NodeManager` embeds the `RWMutex`. This way its API gets public and could be misused. Would prefer it as private field `mu sync.RWMutex`.

**Comment by adambabik** [comment]
This work is tracked in [a swarm](https://github.com/status-im/ideas/blob/master/ideas/63-refactor-geth-packages.md).
